stereo_image_proc/src/stereo_image_proc/disparity_node.cpp

```diff
@@ -326,10 +326,12 @@ DisparityNode::DisparityNode(const rclcpp::NodeOptions & options)
 
         sub_l_image_.subscribe(
           this, left_topic, hints.getTransport(), sensor_data_qos, sub_opts);
-        sub_l_info_.subscribe(this, left_info_topic, sensor_data_qos, sub_opts);
+        sub_l_info_.subscribe(this, left_info_topic,
+          rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(sensor_data_qos)), sub_opts);
```

Comment:
I think we should refactor this a bit.

Above on line 318, we can change sensor_data_qos to:

```
const SensorDataQoS sensor_data_qos;
```

Then in line 327-328, we can change the code to use `get_rmw_qos_profile`:

```
        sub_l_image_.subscribe(
          this, left_topic, hints.getTransport(), sensor_data_qos.get_rmw_qos_profile(), sub_opts);
```

And then finally, on this line, we can do:

```suggestion
        sub_l_info_.subscribe(this, left_info_topic, sensor_data_qos, sub_opts);
```

I think that is a lot more readable than using the `QoSInitialization::from_rmw` method.